tools/criticalstat.py

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/python
+# @lint-avoid-python-3-compatibility-imports
+#
+# criticalstat  Trace long critical sections (IRQs or preemption disabled)
+#               For Linux, uses BCC, eBPF. Requires kernel built with
+#               CONFIG_DEBUG_PREEMPT and CONFIG_PREEMPTIRQ_EVENTS
+#
+# USAGE: criticalstat [-h] [-p] [-i] [-d DURATION]
+#
+# Copyright (c) 2018, Google, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License")
+#
+# By Joel Fernandes <joel@joelfernandes.org>
+
+from __future__ import print_function
+from bcc import BPF
+import argparse
+import ctypes as ct
+import sys
+import subprocess
+import os.path
+
+examples=""
+
+parser = argparse.ArgumentParser(
+    description="Trace long critical sections",
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    epilog=examples)
+
+parser.add_argument("-p", "--preemptoff", action="store_true",
+                    help="Find long sections where preemption was off")
+
+parser.add_argument("-i", "--irqoff", action="store_true",
+                    help="Find long sections where IRQ was off")
+
+parser.add_argument("-d", "--duration", default=100,
+                    help="Duration in uS (microseconds) below which we filter")
+
+args = parser.parse_args()
+
+preemptoff = False
+irqoff = False
+
+if args.irqoff:
+    preemptoff = False
+    irqoff = True
+elif args.preemptoff:
+    preemptoff = True
+    irqoff = False
+
+debugfs_path = subprocess.Popen ("cat /proc/mounts | grep -w debugfs" +
+    " | awk '{print $2}'",
+    shell=True,
+    stdout=subprocess.PIPE).stdout.read().split("\n")[0]
+
+if debugfs_path == "":
+    print("ERROR: Unable to find debugfs mount point");
+    sys.exit(0);
+
+trace_path = debugfs_path + "/tracing/events/preemptirq/";
+
+if (not os.path.exists(trace_path + "irq_disable") or
+   not os.path.exists(trace_path + "irq_enable") or
+   not os.path.exists(trace_path + "preempt_disable") or
+   not os.path.exists(trace_path + "preempt_enable")):
```

Comment:
I like to make code python3 ready. Could you incorporate the below change for python3?
```
diff --git a/tools/criticalstat.py b/tools/criticalstat.py
index 0f90c84..3c133b3 100755
--- a/tools/criticalstat.py
+++ b/tools/criticalstat.py
@@ -51,18 +51,18 @@ elif args.preemptoff:
 debugfs_path = subprocess.Popen ("cat /proc/mounts | grep -w debugfs" +
     " | awk '{print $2}'",
     shell=True,
-    stdout=subprocess.PIPE).stdout.read().split("\n")[0]
+    stdout=subprocess.PIPE).stdout.read().split(b"\n")[0]
 
 if debugfs_path == "":
     print("ERROR: Unable to find debugfs mount point");
     sys.exit(0);
 
-trace_path = debugfs_path + "/tracing/events/preemptirq/";
+trace_path = debugfs_path + b"/tracing/events/preemptirq/";
 
-if (not os.path.exists(trace_path + "irq_disable") or
-   not os.path.exists(trace_path + "irq_enable") or
-   not os.path.exists(trace_path + "preempt_disable") or
-   not os.path.exists(trace_path + "preempt_enable")):
+if (not os.path.exists(trace_path + b"irq_disable") or
+   not os.path.exists(trace_path + b"irq_enable") or
+   not os.path.exists(trace_path + b"preempt_disable") or
+   not os.path.exists(trace_path + b"preempt_enable")):
     print("ERROR: required tracing events are not available\n" +
         "Make sure the kernel is built with CONFIG_DEBUG_PREEMPT " +
```

Reply:
done